src/utils/deprecated/deprecated.tsx

```diff
@@ -3,25 +3,65 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
 
 export const getDeprecatedMessage = (message: string): string =>
   `[DEPRECATED] ${message}`;
 
-export const deprecated = (message: string) => {
+interface IDeprecatedComponentWarning {
+  newComponentName: string;
+  version?: string;
+  getMessage?: (deprecatedComponentName: string) => string;
+}
+
+export const deprecatedComponentWarning = ({
+  newComponentName,
+  version,
+  getMessage,
+}: IDeprecatedComponentWarning) => {
   return <T extends React.ComponentType<any>>(Component: T): T => {
+    const deprecatedComponentName = Component.displayName || Component.name;
+
     const DeprecatedWrapper = (props: React.ComponentProps<T>) => {
       useEffect(() => {
-        console.warn(getDeprecatedMessage(message));
+        const message =
+          getMessage?.(deprecatedComponentName) ||
+          `${deprecatedComponentName} is deprecated in favor of ${newComponentName} and will be removed in v${version}.`;
```

Comment:
`version` is nullable. Might be better to have a message if `version` has no value? Or require a value (but I think the other option would be better in case we don't have a specific version in mind)?